src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -76,16 +77,31 @@ public EditCommand(Index index, EditPersonDescriptor editPersonDescriptor) {
         this.editPersonDescriptor = new EditPersonDescriptor(editPersonDescriptor);
     }
 
+    /**
+     * @param name of the student in the filtered student list to edit
+     * @param editPersonDescriptor details to edit the student with
+     */
+    public EditCommand(Name name, EditPersonDescriptor editPersonDescriptor) {
+        requireNonNull(name);
+        requireNonNull(editPersonDescriptor);
+
+        this.name = name;
+        this.editPersonDescriptor = new EditPersonDescriptor(editPersonDescriptor);
+
+    }
+
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        List<Student> lastShownList = model.getFilteredPersonList();
 
-        if (index.getZeroBased() >= lastShownList.size()) {
+        Student studentToEdit;
+        try {
+            // get the student from model filtered list by either name or index
+            studentToEdit = model.getStudentFromFilteredPersonListByName(name).orElseGet(() ->
```

Comment:
Wow, there's a getStudentFromFilteredPersonListByName method. I think I can use it for my code!!!